packages/shared/useDateFormat/index.ts

```diff
@@ -119,6 +119,6 @@ export type UseDateFormatReturn = ComputedRef<string>
  * @param formatStr - The combination of tokens to format the date
  * @param options - UseDateFormatOptions
  */
-export function useDateFormat(date: MaybeRefOrGetter<DateLike>, formatStr: MaybeRefOrGetter<string> = 'HH:mm:ss', options: UseDateFormatOptions = {}): UseDateFormatReturn {
+export function useDateFormat<T>(date: MaybeRefOrGetter<DateLike>, formatStr: MaybeRefOrGetter<string> = 'HH:mm:ss', options: UseDateFormatOptions = {}): UseDateFormatReturn<T> {
```

Comment:
```suggestion
export function useDateFormat<T = string>(date: MaybeRefOrGetter<DateLike>, formatStr: MaybeRefOrGetter<string> = 'HH:mm:ss', options: UseDateFormatOptions = {}): UseDateFormatReturn<T> {
```
this should default. because it cant be infered. Right?

Reply:
I'm not sure, useDateFormat returns UseDateFormatReturn, and the default value is already set in UseDateFormatReturn.

Reply:
![image](https://github.com/user-attachments/assets/5f9eb30e-9ac0-4dd4-9139-982e3bbd32fc)

https://www.typescriptlang.org/play/?#code/C4TwDgpgBAShwFcBOA7APAFSgXigZ2CQEsUBzAPhygwCg6AzBFAY2CIHsUoE8IAxdu0zkAFAEoAXLHjJ0GcgG8aUKEhmooAIgCGeACaaou6jQC+dZpwL5CVHv0HiaAemcqVAPQD8dUJGmIqABMmFQExGSUuLQMTKwcXPYC7CFYuOEkFOJScIEoqYrKqupcOvqGxrTmNJYo1uFBdrzJQU6u7lDeQA